### PR TITLE
Add network enums

### PR DESCRIPTION
**Pull request by reedsa**
### What was wrong?

Closes ethereum/web3.py#1870

### How was it fixed?

Wrote a [script](https://gist.github.com/reedsa/addafe9eb852db3267c0e90f1ce9867e) to generate the enums file, executed it and committed the result seen here. We can simply rerun the script if we ever need to update these.

### Todo:
- [X] Clean up commit history

- [X] Add or update documentation related to these changes

- [x] Add entry to the [release notes](https://github.com/ethereum/eth-typing/blob/master/newsfragments/README.md)

#### Cute Animal Picture

Put a link to a cute animal picture inside the parenthesis-->

### CODE REVIEW

File: newsfragments/49.feature.rst [hunk 1]
```diff
@@ -0,0 +1 @@
+Enums for Ethereum networks provided by `ChainID` class.
```
Comment (fselmo) [inline newsfragments/49.feature.rst:1]: Same definition here as the description in the comment above.

> Add an `IntEnum` class defining EVM-compatible network name enums as their respective chain id `int` values.

File: docs/eth_typing.rst [hunk 2]
```diff
@@ -49,6 +49,30 @@ Class that contains the different names used to represent hard forks on the Ethe
         Constantinople = 'Constantinople'
         Metropolis = 'Metropolis'
 
+
+ChainId
+~~~~~~~
+
+Class which contains enums for each Ethereum network chain identifier.
```
Comment (fselmo) [inline docs/eth_typing.rst:56]: I'm not sure "Ethereum" fits in this description. If you pulled from `ethereum-lists/chains` then it looks like, according to their documentation, they are at least all EVM compatible. So maybe we go with that verbiage.

Maybe something along these lines? :

```suggestion
`IntEnum` class defining EVM-compatible network name enums as their respective chain id `int` values.
```

File: docs/eth_typing.rst [hunk 3]
```diff
@@ -49,6 +49,30 @@ Class that contains the different names used to represent hard forks on the Ethe
         Constantinople = 'Constantinople'
         Metropolis = 'Metropolis'
 
+
+ChainId
```
Comment (fselmo) [inline docs/eth_typing.rst:53]:
```suggestion
ChainID
```

Or, conversely, I'd support renaming the `IntEnum` class `ChainId`

File: newsfragments/49.feature.rst (hunk 1 above)
Comment (reedsa) [reply newsfragments/49.feature.rst:1]: Oops, keep getting this backwards. Will double check in the future!

File: docs/eth_typing.rst (hunk 3 above)
Comment (reedsa) [reply docs/eth_typing.rst:53]: Renamed the enum, feels closer to convention.

**Review by fselmo — APPROVED** [review]
lgtm 🚀

**Review by pacrob — APPROVED** [review]
lgtm!

**Pull request merged**